src/kernels/jupyter/launcher/serverUriStorage.ts

```diff
@@ -122,7 +130,9 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage, IServe
             blob
         );
     }
-    public async getSavedUriList(): Promise<{ uri: string; time: number; displayName?: string | undefined }[]> {
+    public async getSavedUriList(): Promise<
+        { uri: string; serverId: string; time: number; displayName?: string | undefined }[]
```

Comment:
Instead of having to get the server id for the saved uri list all over the place, I just changed this to return it too.